client/helpers/map-domain-description.spec.js

```diff
@@ -0,0 +1,185 @@
+    it('should return "description" = "DomainDescription".', () => {
+      const domain = {
+        domainInfo: {
+          description: 'DomainDescription',
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output.description).toEqual('DomainDescription');
+    });
+  });
+
+  describe('When domain.domainInfo.ownerEmail = "OwnerEmail"', () => {
+    it('should return "owner" = "OwnerEmail".', () => {
+      const domain = {
+        domainInfo: {
+          ownerEmail: 'OwnerEmail',
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output.owner).toEqual('OwnerEmail');
+    });
+  });
+
+  describe('When domain.isGlobalDomain = true', () => {
+    it('should return "Global?" = "Yes".', () => {
+      const domain = {
+        isGlobalDomain: true,
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['Global?']).toEqual('Yes');
+    });
+  });
+
+  describe('When domain.configuration.workflowExecutionRetentionPeriodInDays = 3', () => {
+    it('should return "Retention Period" = "3 days".', () => {
+      const domain = {
+        configuration: {
+          workflowExecutionRetentionPeriodInDays: 3,
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['Retention Period']).toEqual('3 days');
+    });
+  });
+
+  describe('When domain.configuration.emitMetric = true', () => {
+    it('should return "Emit Metrics" = "Yes".', () => {
+      const domain = {
+        configuration: {
+          emitMetric: true,
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['Emit Metrics']).toEqual('Yes');
+    });
+  });
+
+  describe('When domain.configuration.historyArchivalStatus = "ENABLED"', () => {
+    it('should return "History Archival" = "Enabled".', () => {
+      const domain = {
+        configuration: {
+          historyArchivalStatus: 'ENABLED',
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['History Archival']).toEqual('Enabled');
+    });
+  });
+
+  describe('When domain.configuration.visibilityArchivalStatus = "ENABLED"', () => {
+    it('should return "Visibility Archival" = "Enabled".', () => {
+      const domain = {
+        configuration: {
+          visibilityArchivalStatus: 'ENABLED',
+        },
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['Visibility Archival']).toEqual('Enabled');
+    });
+  });
+
+  describe('When domain.failoverVersion = 1', () => {
+    it('should return "Failover Version" = 1', () => {
+      const domain = {
+        failoverVersion: 1,
+      };
+
+      const output = mapDomainDescription(domain);
+
+      expect(output['Failover Version']).toEqual(1);
+    });
+  });
+
+  describe(`
+    When domain.replicationConfiguration.clusters = [{ clusterName: "cluster1" }, { clusterName: "cluster2" }]
```

Comment:
For some of these long descriptions, is there a more human-readable way to write them? Currently most of the test descriptions seem very programatic and not a great summary of expected behavior.

For example, would this make sense 'Multiple clusters with one active cluster'?

Reply:
I can update. Thanks